src/Polly.Shared/Retry/Alternative.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+
+namespace Polly.Retry
+{
+    /// <summary>
+    /// </summary>
+    public static class Alternative // Experiment
```

Comment:
This is the alternative API shape. 
* We'd need a static class that holds the methods, called say 'SleepDurationSeriesStrategy`. 
* We would delete 'ISleepDurationSeriesStrategy` and all its concrete classes, since the `interface` would have no purpose (it would be a marker interface only)
* We would add the different sleep duration methods shown below.

Pros: Less cruft
Cons: Less encapsulation

(Please focus on api shape for now, not on comments, etc)

Reply:
@grant-d Great to see 👍 .  I was thinking that all the configuration parameters to generate the `IEnumerable<TimeSpan>` should be configured together (not some in the ctor, some in a method), which essentially gives a choice between forms like:

(1) `new DecorellatedJitter(param1, param2, retryCount).GetSleepDurations()` // pre-configured instance with a no-params get method
(2) `DecorellatedJitter.GetSleepDurations(param1, param2, retryCount)` // static factory method on static class-per-strategy  .  <sub>Same as your `SleepDurationSeriesStrategy` factory class, except not overloading all variants on a single factory class</sub>

Community views on advs / disadvs either way welcome.  

Polly configuration has tended to take a static-factory-method declarative approach like (2), so my initial instinct lies here.  If a given set of parameters isn't re-used, (2) also reads more concisely than (1) with its extra `new`.  But an instance approach (1) could bind better with the .NET Core `IOptions<>` approach to config binding.  

A disadvantage of the instance approach (1) could be if it tempts / inadvertently guides users to a singleton `DecorrelatedJitter` instance (or single `ISleepDurationSeriesStrategy` fulfilment instance) to use all around their app. Singleton configuration can be a good starting point but isn't always a good fit for real world infrastructure - it's common to have a mix of more reliable/nearby (in network terms) resources like eg other Azure resources, versus more fragile eg external legacy systems, which might need different strategies.  And the absence of named-registrations in classic Net Core DI makes multiple variant registrations of the same class/interface hard.  Then again, this could be covered by documentation.  And a _recommended_ / more typical Polly pattern is to configure all policies at startup, so it would be less usual (tho possible) to be DI-injecting `DecorellatedJitter` instances into point-of-use classes, to configure a policy there.  

Community views?